kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/parser/feel11/ParserHelper.java

```diff
@@ -108,16 +114,63 @@ public void recoverScope( String name ) {
+                pushScope(resolvedBIType);
+                switch (resolvedBIType) {
+                    // FEEL spec table 53
+                    case DATE:
+                        this.currentScope.define(new VariableSymbol( "year", BuiltInType.NUMBER ));
+                        this.currentScope.define(new VariableSymbol( "month", BuiltInType.NUMBER ));
+                        this.currentScope.define(new VariableSymbol( "day", BuiltInType.NUMBER ));
+                        break;
+                    case TIME:
+                        this.currentScope.define(new VariableSymbol( "hour", BuiltInType.NUMBER ));
+                        this.currentScope.define(new VariableSymbol( "minute", BuiltInType.NUMBER ));
+                        this.currentScope.define(new VariableSymbol( "second", BuiltInType.NUMBER ));
+                        this.currentScope.define(new VariableSymbol( "time offset", BuiltInType.NUMBER ));
+                        this.currentScope.define(new VariableSymbol( "timezone", BuiltInType.NUMBER ));
+                        break;
+                    case DATE_TIME:
+                        this.currentScope.define(new VariableSymbol( "year", BuiltInType.NUMBER ));
+                        this.currentScope.define(new VariableSymbol( "month", BuiltInType.NUMBER ));
+                        this.currentScope.define(new VariableSymbol( "day", BuiltInType.NUMBER ));
+                        this.currentScope.define(new VariableSymbol( "hour", BuiltInType.NUMBER ));
+                        this.currentScope.define(new VariableSymbol( "minute", BuiltInType.NUMBER ));
+                        this.currentScope.define(new VariableSymbol( "second", BuiltInType.NUMBER ));
+                        this.currentScope.define(new VariableSymbol( "time offset", BuiltInType.NUMBER ));
+                        this.currentScope.define(new VariableSymbol( "timezone", BuiltInType.NUMBER ));
+                        break;
+                    case DURATION:
+                        // TODO might need to distinguish between `years and months duration` and `days and time duration`
+                        this.currentScope.define(new VariableSymbol( "years", BuiltInType.NUMBER ));
+                        this.currentScope.define(new VariableSymbol( "months", BuiltInType.NUMBER ));
+                        this.currentScope.define(new VariableSymbol( "days", BuiltInType.NUMBER ));
+                        this.currentScope.define(new VariableSymbol( "hours", BuiltInType.NUMBER ));
+                        this.currentScope.define(new VariableSymbol( "minutes", BuiltInType.NUMBER ));
+                        this.currentScope.define(new VariableSymbol( "seconds", BuiltInType.NUMBER ));
+                        break;
+                    // table 53 applies only to type(e) is a date/time/duration
+                    case UNKNOWN:
+                        enableDynamicResolution();
+                        break;
```

Comment:
this is because if a type is composed in an "anonymous" context, we don't have a DMN representation as `ItemDefinition` of that type hence it was originally just "unknown" but then in could be used in a subsequent context entry, there is an isolated test case added to highlight this.